### PR TITLE
ref(charts): Move xAxis date formatting out of `<ChartZoom>`

### PR DESCRIPTION
**Pull request by billyvg**
Moved this into `<BaseChart>` and `XAxis` helper so that all charts can get this logic

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/components/xAxis.jsx [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+  end,
+  period,
   utc,
   ...props
 } = {}) {
-  const axisLabelFormatter = value => {
+  const axisLabelFormatter = (value, index) => {
     if (isGroupedByDate) {
-      const format = shouldRenderTimeOnly === 'hour' ? 'LT' : 'MMM Do';
+      const dateFormat = useShortDate ? 'MMM Do' : 'MMM D LT';
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/charts/components/xAxis.jsx]: Having this here will is great. Will help make all our charts more consistent. Should we also take the user's clock preferences into account when formatting axis labels?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/charts/components/xAxis.jsx]: Yeah, I'll do followups for clock and maybe tz preferences as well.

**Comment by billyvg** [comment]
Feel free to click around https://sentry-git-ref-chartsmove-xaxis-logic-from-chart-zoom.sentry.now.sh/

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Pull request merged**